### PR TITLE
Keep remainder in fee account + small bug fix

### LINKED ISSUE #8129: feat: liquidity fee remainder carried over
# Feature Overview

Flooring liquidity fees owed to each LP can result in a remainder in the liquidity pool after distribution. Currently this remainder is transferred to the last LP in the list. This can result in an LP with a small commitment receiving more fees than an LP with a larger commitment. See this draft [feature test](https://github.com/vegaprotocol/vega/pull/8102/files).

LP fee distribution should be updated so any remainder is left in the liquidity fee pool until the next fee distribution period. If fees are being distributed for the final time as part of market settlement the remainder should be transferred to the network treasury for that settlement asset.

Change will break a number of tests which check fee distribution amounts and possibly tests which settle markets and check the balance of the network treasury. 

## Specs
- [0042-LIQF](https://github.com/vegaprotocol/specs/pull/1655) updated to reflect proposed change.

# Tasks
A checklist of the tasks that are needed to develop the feature and meet the acceptance criteria and feature test scenarios. Ideally, tasks would reflect the issues/pull requests likely to be created when developing the feature. 
- Update liquidity fee distribution mechanisms.
- Fix broken tests.

# Acceptance Criteria
- 0042-LIQF-031
- 0042-LIQF-032

# Test Scenarios
Given this [feature test](https://github.com/vegaprotocol/vega/pull/8102/files), the smallest LP should no longer receive a fee and the balance in the liquidity fee pool should not be zero after distribution.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Create relevant for [system-test](https://github.com/vegaprotocol/system-tests/issues) tickets with feature labels
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All acceptance criteria confirmed to be met, or, reasons why not discussed with the engineering leadership team
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new system-tests
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by EVODelavega**
Closes #8129 

In the process, noticed a small bug where an empty insurance pool wasn't properly removed (and thus would be part of insurance funds distribution when another market terminated).

### CODE REVIEW

File: core/collateral/engine.go [hunk 1]
```diff
@@ -2096,12 +2096,31 @@ func (e *Engine) ClearMarket(ctx context.Context, mktID, asset string, parties [
...
+		resps = append(resps, lpFeeLE)
+		// remove the account once it's drained
+		e.removeAccount(lpFeeAccID)
+	}
 
 	// redistribute the remaining funds in the market insurance account between other markets insurance accounts and global insurance account
-	marketInsuranceID := e.accountID(mktID, "", asset, types.AccountTypeInsurance)
-	marketInsuranceAcc, ok := e.accs[marketInsuranceID]
-	if !ok || marketInsuranceAcc.Balance.EQ(num.UintZero()) {
+	if marketInsuranceAcc.Balance.EQ(num.UintZero()) {
```
Comment (ze97286) [inline core/collateral/engine.go:2121]: `if marketInsuranceAcc.Balance.IsZero()`

Comment (EVODelavega) [reply core/collateral/engine.go]: done

File: core/fee/engine.go [hunk 2]
```diff
@@ -424,20 +425,16 @@ func (e *Engine) BuildLiquidityFeeDistributionTransfer(shares map[string]num.Dec
...
+				Amount: amount,
 				Asset:  acc.Asset,
 			},
 			MinAmount: amount.Clone(),
 			Type:      types.TransferTypeLiquidityFeeDistribute,
 		})
 	}
 
+	// if there is a remainder, just keep it in the fee account, will be used next time we pay out fees
 	// last is the party who will get the remaining from ceil
```
Comment (ze97286) [inline core/fee/engine.go:437]: this is no longer true I reckon.

**Review by ze97286 — DISMISSED** [review]

**Review by cdummett — DISMISSED** [review]

File: core/fee/engine.go (hunk 2 above)
Comment (EVODelavega) [reply core/fee/engine.go:437]: Yes it is, that's the whole point of the change

**Review by ze97286 — DISMISSED** [review]

**Review by jeremyletang — APPROVED** [review]

**Review by cdummett — APPROVED** [review]

**Pull request merged**